modules/partition-replicator/src/test/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManagerTest.java

```diff
@@ -324,4 +378,55 @@ void producesEventsOnPartitionDestroy() {
         assertThat(beforeReplicaDestroyedFuture.thenApply(LocalPartitionReplicaEventParameters::zonePartitionId), willBe(zonePartitionId));
         assertThat(afterReplicaDestroyedFuture.thenApply(LocalPartitionReplicaEventParameters::zonePartitionId), willBe(zonePartitionId));
     }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenAllComponentsAreStoppedFine() throws Exception {
+        assertDoesNotThrow(() -> partitionReplicaLifecycleManager.beforeNodeStop());
+
+        assertThat(partitionReplicaLifecycleManager.stopAsync(), willCompleteSuccessfully());
+
+        verify(replicaManager, times(CatalogUtils.DEFAULT_PARTITION_COUNT)).stopReplica(any());
+    }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenReplicasAreStoppedExceptionally() throws Exception {
+        when(replicaManager.stopReplica(any())).thenThrow(new NodeStoppingException());
+
+        assertDoesNotThrow(() -> partitionReplicaLifecycleManager.beforeNodeStop());
+
+        assertThat(partitionReplicaLifecycleManager.stopAsync(), willCompleteSuccessfully());
+
+        // One more because of exceptional mock above counts.
+        verify(replicaManager, times(CatalogUtils.DEFAULT_PARTITION_COUNT + 1)).stopReplica(any());
+
+        // Do reset for correct replica manager stop on tear down.
+        reset(replicaManager);
+    }
+
+    @Test
+    public void partitionLifecycleManagerStopsCorrectWhenTxStatePartitionStoragesAreStoppedExceptionally() throws Exception {
+        doReturn(commonZonePartitionResources).when(zoneResourcesManager).getZonePartitionResources(any());
+
+        int defaultZoneId = 0;
+        List<ZonePartitionResources> defaultZoneResources = IntStream.range(0, CatalogUtils.DEFAULT_PARTITION_COUNT)
+                .mapToObj(partId -> new ZonePartitionId(defaultZoneId, partId))
+                .map(partitionReplicaLifecycleManager::zonePartitionResources)
+                .collect(Collectors.toList());
+
+        defaultZoneResources.forEach(resources -> {
+            TxStatePartitionStorage txStorage = resources.txStatePartitionStorage();
+
+            doThrow(new RuntimeException()).when(txStorage).close();
+        });
+
+        assertDoesNotThrow(() -> partitionReplicaLifecycleManager.beforeNodeStop());
+
+        assertThat(partitionReplicaLifecycleManager.stopAsync(), willThrow(RuntimeException.class));
+
+        verify(replicaManager, times(CatalogUtils.DEFAULT_PARTITION_COUNT)).stopReplica(any());
+
+        defaultZoneResources.forEach(resources -> verify(resources.txStatePartitionStorage(), atLeastOnce()).close());
+
+        defaultZoneResources.forEach(resources -> reset(resources.txStatePartitionStorage()));
```

Comment:
What is this about?

Reply:
I was afraid that it may affects on `tearDown` like replica manager did without it's reset in `partitionLifecycleManagerStopsCorrectWhenReplicasAreStoppedExceptionally`. But looks like its overly.